src/lib.rs

```diff
@@ -263,264 +213,322 @@ where
         disk_chunk_size: usize,
         item_buffer_size: usize,
     ) -> Result<ShardWriter<T, S>, Error> {
-        let (send, recv) = bounded(4);
-
-        // Ping-pong of the 2 item buffer between the thread the accumulates the items,
-        // and the thread that does the IO. Need 2 slots to avoid a deadlock.
-        let (to_writer_send, to_writer_recv) = bounded(2);
-        let (to_buffer_send, to_buffer_recv) = bounded(2);
-
-        // Divide buffer size by 2 -- the get swaped between buffer thread and writer thread
-        let mut sbt = ShardBufferThread::<T>::new(
-            sender_buffer_size,
-            item_buffer_size >> 1,
-            recv,
-            to_writer_send,
-            to_buffer_recv,
-        );
-        let p2 = path.as_ref().to_owned();
-
-        let buffer_thread = thread::spawn(move || sbt.process());
-
-        let writer = FileManager::<<S as SortKey<T>>::Key>::new(p2)?;
-        let writer_thread = thread::spawn(move || {
-            let mut swt = ShardWriterThread::<_, S>::new(
-                disk_chunk_size,
-                writer,
-                to_writer_recv,
-                to_buffer_send,
-            );
-            swt.process()
-        });
+        assert!(disk_chunk_size >= 1);
+        assert!(item_buffer_size >= 1);
+        assert!(sender_buffer_size >= 1);
```

Comment:
Should we also assert for the relative values? Like `assert!(item_buffer_size > sender_buffer_size)`?

Reply:
good idea, added.